x-pack/plugins/siem/server/lib/hosts/elasticsearch_adapter.ts

```diff
@@ -74,3 +100,36 @@ export const formatHostsData = (
       },
     }
   );
+
+const formatHostItem = (fields: ReadonlyArray<string>, bucket: HostAggEsItem): HostItem =>
+  fields.reduce<HostItem>((flattenedFields, fieldName) => {
+    const fieldValue = getHostFieldValue(fieldName, bucket);
+    if (fieldValue) {
+      return set(fieldName, fieldValue, flattenedFields);
+    }
+    return flattenedFields;
+  }, {});
+
+const getHostFieldValue = (fieldName: string, bucket: HostAggEsItem): string | string[] | null => {
+  const aggField = hostFieldsMap[fieldName]
+    ? hostFieldsMap[fieldName].replace(/\./g, '_')
+    : fieldName.replace(/\./g, '_');
+  if (['host.ip', 'host.mac'].includes(fieldName) && has(aggField, bucket)) {
+    const data: HostBuckets = get(aggField, bucket);
+    return data.buckets.map(obj => obj.key);
+  } else if (has(`${aggField}.buckets`, bucket)) {
+    return getFirstItem(get(`${aggField}`, bucket));
+  } else if (has(aggField, bucket)) {
+    const valueObj: HostValue = get(aggField, bucket);
+    return valueObj.value_as_string;
+  }
+  return null;
+};
+
+const getFirstItem = (data: HostBuckets): string | null => {
+  const firstItem = head(data.buckets);
+  if (firstItem == null) {
+    return null;
+  }
+  return firstItem.key;
+};
```

Comment:
These are all nice pure functions that would be good candidates for unit tests

Reply:
they are tested globally through the elasticsearch_adapter.test.ts